flink-tests/src/test/java/org/apache/flink/test/runtime/leaderelection/ZooKeeperLeaderElectionITCase.java

```diff
@@ -141,13 +144,23 @@ private DispatcherGateway getNextLeadingDispatcherGateway(TestingMiniCluster min
 		return miniCluster.getDispatcherGatewayFuture().get();
 	}
 
-	private JobGraph createJobGraph(int parallelism) {
+	private JobGraph createJobGraph(int parallelism) throws IOException {
 		BlockingOperator.isBlocking = true;
 		final JobVertex vertex = new JobVertex("blocking operator");
 		vertex.setParallelism(parallelism);
 		vertex.setInvokableClass(BlockingOperator.class);
 
-		return new JobGraph("Blocking test job", vertex);
+		JobGraph jobGraph = new JobGraph("Blocking test job", vertex);
+
+		// explicitly allow restarts; this is necessary since the shutdown may result in the job failing and hence being
+		// removed from ZooKeeper. What happens to running jobs if the Dispatcher shuts down in an orderly fashion
+		// is undefined behavior. By allowing restarts we prevent the job from reaching a globally terminal state,
+		// causing it to be recovered by the next Dispatcher.
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		executionConfig.setRestartStrategy(RestartStrategies.fixedDelayRestart(10, Duration.ofSeconds(10).toMillis()));
```

Comment:
```suggestion
		executionConfig.setRestartStrategy(RestartStrategies.fixedDelayRestart(100, Duration.ofMillis(100).toMillis()));
```

Reply:
My idea here was to not have the job actually restart since it shouldn't be relevant to the test whether the job is running/restarting (just that it's not failed), and these state transitions add additional noise to the logs.

Reply:
Good point. Then ignore my comment.